subsys/bluetooth/gatt_dm.c

```diff
@@ -290,9 +290,13 @@ static uint8_t discovery_process_service(struct bt_gatt_dm *dm,
 		return BT_GATT_ITER_STOP;
 	}
 
-	LOG_DBG("Service detected, handles range: <%u, %u>",
-		cur_attr->handle + 1,
-		service_val->end_handle);
+	if (cur_attr->handle == service_val->end_handle) {
+		LOG_DBG("Empty service detected with handle: %u", cur_attr->handle);
```

Comment:
What happens when we don't return here. It looks like there is a risk that we will start  attribute discovery with invalid handle range. Am I right or I missed something ?